contributors/design-proposals/scheduling/coscheduling.md

```diff
@@ -0,0 +1,174 @@
+
+## Function Detail
+
+### API Definition
+
+The following requirements are identified during the discussion of this feature:
+
+1. Existing workload can use this feature without (or with a few) configuration changes
+2. Pods of a group/gang may have different `PodSpec` (and/or belong to different collections)
+3. Existing controllers which are responsible for managing life cycle of collections work well with this feature
+
+To meet the requirements above, the following **Kind** is introduced in core part under `scheduling.k8s.io/v1alpha1` **Group**/**Version**.
+
+```go
+// PodGroup defines the scheduling requirement of a pod group
+type PodGroup struct {
+    metav1.TypeMeta
+    metav1.ObjectMeta
+
+	// Spec defines the behavior of a pod group.
+	// +optional
+    Spec PodGroupSpec
+
+	// Status represents the current information about a pod group.
+	// This data may not be up to date.
+	// +optional
+    Status PodGroupStatus
+}
+
+// PodGroupSpec represents the template of a pod group.
+type PodGroupSpec struct {
+    // NumMember defines the number of members/tasks to run the pod group;
+    // if there's not enough resources to start all tasks, the scheduler
+    // will not start anyone.
+    NumMember int
+
+    // Resources defines the total resource the PodGroup requests to run
+    // Pods.
+    Resources v1.ResourceList
+}
+
+// PodGroupStatus represents the current state of a pod group.
+type PodGroupStatus struct {
+    // The number of actively running pods.
+    // +optional
+    Running int32
+
+    // The number of pods which reached phase Succeeded.
+    // +optional
+    Succeeded int32
+
+    // The number of pods which reached phase Failed.
+    // +optional
+    Failed int32
+}
+```
+
+The `PodGroup`, which is a namespaced object, specifies the behavior and status of a pod group, e.g. number of pods in a group. To define which pods are member of `PodGroup`, the following annotation key is introduced for `Pod`.
+
+```go
+scheduling.k8s.io/group-name
+```
+
+The `scheduling.k8s.io/group-name` annotation specifies the `PodGroup` that it belongs to; and the pod can only belong to the `PodGroup` in the same namespace. The pod, controlled by different collections, can also belong to the same `PodGroup`.  Because of performance concern,  it does not use `LabelSelector` to build the relationship between `PodGroup` and `Pod`.
+
+### Lifecycle Management
+
+As the lifecycle of Pods in PodGroup maybe different from controller to controller, the lifecycle management is delegated to the owner of Pods. In kube-arbitrator, it'll record related events for controller to manage pods, e.g. `Unschedulable`.
+
+The update to `PodGroup` is not support for now; and deleting `PodGroup` does not impact Pod's status.
+
+### kube-arbitrator
```

Comment:
What is kube-arbitrator?

Reply:
From reading below I see `kube-arbitrator` means an alternate scheduler in this construct. Given that there is general agreement to promote this feature to default scheduler in the future, should it be added to the default scheduler as an experimental (disabled by default) feature now?

Reply:
> should it be added to the default scheduler as an experimental (disabled by default) feature now?

We'd like to use "scheduler framework" to handle that; and Coscheduling/Gang-scheduling just one requirements for batch workload. I'd like to have a separated project to introduce "job" concept in scheduler (to work with scheduler framework) and collect enough requirements.

BTW, we incubate `kube-arbitrator` because of complexity of "batch" :)